base/broadcast.jl

```diff
@@ -431,57 +510,177 @@ julia> (1 + im) ./ Nullable{Int}()
 Nullable{Complex{Float64}}()
 ```
 """
-@inline broadcast(f, A, Bs...) = broadcast_c(f, containertype(A, Bs...), A, Bs...)
+@inline broadcast(f, A, Bs...) =
+    broadcast(f, Result{combine_types(A, Bs...)}(), A, Bs...)
+
+"""
+    broadcast(f, Broadcast.Result{ContainerType}(), As...)
```

Comment:
I dislike asking people to specialize either `broadcast` or `_broadcast`. I think I'll just ditch the docstring and hope people will find the info they need in the manual.

Reply:
Sounds good. I guess you could add a link to the manual in the generic docstring for `broadcast` (with a short sentence at the end). Understanding how that complex system works can be useful even for people who don't want to override it.

Reply:
On the other hand, having the mechanisms for broadcast customization be readily discoverable is great. So as an alternative resolution, perhaps retain the docstring and clearly mark it as developer-oriented?

Reply:
It's unlikely that somebody writing a custom implementation of the `broadcast` machinery could do it without reading the manual, so is it really useful to show it in addition to a link to the manual?

Reply:
(Working from an outdated page, my response was meant to follow Tim's rather than offer an alternative to your later suggestion.) Having a sentence regarding broadcast specialization/customization and an associated link to the corresponding manual section in the generic broadcast docstring sounds just as well to me :).